services/observation_helper.py

```diff
@@ -29,12 +30,22 @@ def add_observation(session: Session, data: BaseModel) -> Base:
     #     if 'sample_id' not in data:
     #         sample = Sample(thing_id=thing_id,
     #                         collection_method=data.get('collection_method', 'manual'),
-    #                         collection_timestamp=data.get('observation_timestamp'))
+    #                         collection_timestamp=data.get('observation_datetime'))
     #         session.add(sample)
     #         data['sample'] = sample
     #     else:
     #         raise ValueError('Cannot specify both thing_id and sample_id')
-
+    if "field_sample_id" in data:
```

Comment:
I think that this if block can be removed because

- The `observation` model and schema don't have a `field_sample_id` field (this may be an artifact from a previous iteration of the schema?)
- `sample_id` is also a non-nullable field with a foreign key constraint, so it should be in the payload (not popped).
  - If the corresponding `sample` does not exist the database will raise an error and that can be handled at the endpoint (raise a 409 error with an appropriate error message).